### PR TITLE
Personalized Head

### PR DESCRIPTION
**Pull request by chandralegend**
## Describe your changes
This PR introduces a new feature called 'PH: Personalized Head'. Using the Personalized Head module, you can create a custom model head that you can train over time. You can use your own custom models and datasets to create a personalized head using a configuration file and a python file.

## Will this impact Jaseci users? If so, please write a paragraph describing the impact.
By using this feature, jaseci users are now allowed to use their own custom models with their jaseci app without going through the hassle of writing lots of boilerplate code to make their model work with jaseci. Meanwhile, with PH and Compositor(Not Implemented Yet), jaseci users can allow users to have their own small personalized model without wasting important resources.

#README Consists of a Tutorial(Kinda) on how to make MNIST Classification Model in jaseci from scratch.

### CODE REVIEW

**Comment by ypkang** [comment]
Task linked: [CU-3e6fg0c Personalized Head Python Implementation](https://app.clickup.com/t/3e6fg0c)

**Comment by chandralegend** [comment]
@ypkang Jaseci Core Tests are failing after the linting and pre commit

**Comment by chandralegend** [comment]
@ypkang Ready to Merge

**Comment by ypkang** [comment]
@chandralegend For some reason, this PR is touching almost all the markdown files in the repo. It looks like it is removing a trailing white space at the end of every file. I am not sure if this is intendend by the BLACK formatter or the opposite. Can you check?

Just want to make sure we are not introducing a big commit for no reason.

**Comment by chandralegend** [comment]
@ypkang it happened after pre-commit

File: jaseci_ai_kit/jaseci_ai_kit/modules/ph/README.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+actions load module jaseci_kit.ph
+```
+
+- Run the JAC program using the command below.
+
+```bash
+jac run main.jac -ctx '{"input_image": "test.jpg"}'
+```
+
+## Todo
```
Comment (ypkang) [inline jaseci_ai_kit/jaseci_ai_kit/modules/ph/README.md:341]: @chandralegend do these Todo items still apply?

File: jaseci_ai_kit/jaseci_ai_kit/modules/ph/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+torchvision
```
Comment (ypkang) [inline jaseci_ai_kit/jaseci_ai_kit/modules/ph/requirements.txt]: Do we actually need torchvision and torchaudio? Just trying to make sure we don't blow up the dependency list of jaseci_ai_kit. Also, do we need pytorch-cuda? 

If these are must-haves, can you add them to the setup.py under jaseci_ai_kit? so they will get installed when you run pip install jaseci_ai_kit

**Review by ypkang — CHANGES_REQUESTED** [review]

File: jaseci_ai_kit/jaseci_ai_kit/modules/ph/README.md (hunk 1 above)
Comment (chandralegend) [reply jaseci_ai_kit/jaseci_ai_kit/modules/ph/README.md:341]: yes

File: jaseci_ai_kit/jaseci_ai_kit/modules/ph/requirements.txt (hunk 2 above)
Comment (chandralegend) [reply jaseci_ai_kit/jaseci_ai_kit/modules/ph/requirements.txt]: they are not needed, removed them

**Review by ypkang — APPROVED** [review]
LGTM. Except this adds torchvision to the jaseci_ai_kit dependency which I don't it's needed. I will merge this and then do a tiny followup to fix it.

**Pull request merged**